File: src/main/java/org/acg/user_services/repository/RoleRepo.java
package org.acg.user_services.repository;

import org.acg.user_services.repository.entity.Role;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface RoleRepo extends CrudRepository<Role, Long> {
    Role findByName(String name);
}

File: src/main/kotlin/org/acg/user/repository/RoleRepo.kt
package org.acg.user.repository;

import org.acg.user.repository.entity.Role;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
interface RoleRepo : CrudRepository<Role, Long> {
    fun findByName(name: String): Role
}